lib/_http_agent.js

```diff
@@ -168,6 +168,7 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     var socket = this.freeSockets[name].shift();
     // Assign the handle a new asyncId and run any init() hooks.
     socket._handle.asyncReset();
+    socket[async_id_symbol] = socket._handle.getAsyncId();
```

Comment:
Hmm … doing it this way doesn’t introduce any new methods that are accessible to the outside world. I am okay with moving it, but that should be based on more consideration than just fixing this bug; as you pointed out in the issue, we would want to consider what kind of API, if any, custom Agent implementations need.

So: I’m not disagreeing, but I think it’s better to postpone that after landing this.